Source/FolioReaderContainer.swift

```diff
@@ -72,131 +89,135 @@ open class FolioReaderContainer: UIViewController {
             kCurrentScrollDirection: FolioReaderScrollDirection.defaultVertical.rawValue
             ])
     }
-    
-    /**
-     Set the `FolioReaderConfig` and epubPath.
-     
-     - parameter config:     A instance of `FolioReaderConfig`
-     - parameter path:       The ePub path on system
-     - parameter removeEpub: Should delete the original file after unzip? Default to `true` so the ePub will be unziped only once.
-     */
+
+    /// Set the `FolioReaderConfig` and epubPath.
+    ///
+    /// - Parameters:
+    ///   - config: Current Folio Reader configuration
+    ///   - path: The ePub path on system. Must not be nil nor empty string.
+    ///   - removeEpub: Should delete the original file after unzip? Default to `true` so the ePub will be unziped only once.
     open func setupConfig(_ config: FolioReaderConfig, epubPath path: String, removeEpub: Bool = true) {
-        readerConfig = config
-        epubPath = path
-        shouldRemoveEpub = removeEpub
+        self.readerConfig = config
+        self.folioReader = FolioReader()
+        self.folioReader.readerContainer = self
+        self.epubPath = path
+        self.shouldRemoveEpub = removeEpub
+
+        // Set the shared instance to support old version.
+        FolioReader.shared = self.folioReader
     }
-    
+
     // MARK: - View life cicle
-    
+
     override open func viewDidLoad() {
         super.viewDidLoad()
-        
-        readerConfig.canChangeScrollDirection = isDirection(readerConfig.canChangeScrollDirection, readerConfig.canChangeScrollDirection, false)
-        
+
+        let canChangeScrollDirection = self.readerConfig.canChangeScrollDirection
+        self.readerConfig.canChangeScrollDirection = self.readerConfig.isDirection(canChangeScrollDirection, canChangeScrollDirection, false)
+
         // If user can change scroll direction use the last saved
-        if readerConfig.canChangeScrollDirection {
-            var scrollDirection = FolioReaderScrollDirection(rawValue: FolioReader.currentScrollDirection) ?? .vertical
+        if (self.readerConfig.canChangeScrollDirection == true) {
+            var scrollDirection = (FolioReaderScrollDirection(rawValue: self.folioReader.currentScrollDirection) ?? .vertical)
 
-            if (scrollDirection == .defaultVertical && readerConfig.scrollDirection != .defaultVertical) {
-                scrollDirection = readerConfig.scrollDirection
+            if (scrollDirection == .defaultVertical && self.readerConfig.scrollDirection != .defaultVertical) {
+                scrollDirection = self.readerConfig.scrollDirection
```

Comment:
I am not sure what you mean there. I know that style guide, and I also know that we do not follow it.
We actually branched off from that one and wrote our own.
You can find it there: https://github.com/kevindelord/swift-style-guide

As we would like to merge this code change into @hebertialmeida's repo, we will of course follow/do what you/him want(s) us to modify.